azure_functions_worker/dispatcher.py

```diff
@@ -334,33 +329,54 @@ async def _handle__functions_metadata_request(self, request):
         logger.info(
-            'Received WorkerMetadataRequest, request ID %s, function_path: %s',
-            self.request_id, function_path)
-
+            'Received WorkerMetadataRequest, request_ID %s, function_path: %s, '
+            'File_exists_and_correctly_named: %s',
+            self.request_id, function_path, v2_file_exist)
         try:
-            validate_script_file_name(script_file_name)
-
-            if not os.path.exists(function_path):
-                # Fallback to legacy model
-                logger.info("%s does not exist. "
-                            "Switching to host indexing.", script_file_name)
+            if v2_file_exist and file_name_valid:
+                # V2 programming model
+                fx_metadata_results = self.index_functions(function_path)
+                return protos.StreamingMessage(
+                    request_id=request.request_id,
+                    function_metadata_response=protos.FunctionMetadataResponse(
+                        function_metadata_results=fx_metadata_results,
+                        result=protos.StatusResult(
+                            status=protos.StatusResult.Success)))
+            elif not v2_file_exist and file_name_valid:
+                # V1 programming model
                 return protos.StreamingMessage(
                     request_id=request.request_id,
                     function_metadata_response=protos.FunctionMetadataResponse(
                         use_default_metadata_indexing=True,
                         result=protos.StatusResult(
                             status=protos.StatusResult.Success)))
-
-            fx_metadata_results = self.index_functions(function_path)
-
-            return protos.StreamingMessage(
-                request_id=request.request_id,
-                function_metadata_response=protos.FunctionMetadataResponse(
-                    function_metadata_results=fx_metadata_results,
-                    result=protos.StatusResult(
-                        status=protos.StatusResult.Success)))
-
+            elif v2_file_exist and not file_name_valid:
+                # V2 programming model but with invalid file name
+                error_logger.error('Invalid script file name.'
+                                   'Script File Name %s:', script_file_name)
+                return protos.StreamingMessage(
+                    request_id=request.request_id,
+                    function_metadata_response=protos.FunctionMetadataResponse(
+                        result=protos.StatusResult(
+                            status=protos.StatusResult.Failure,
+                            exception=self._serialize_exception(
+                                Exception('Invalid Script file name %s',
+                                          script_file_name)))))
+            else:
+                logger.error('App setting %s detected but file path %s does '
```

Comment:
For this case - how did we detect the app setting explicitly set? Because the current issue is using the default and that'll always return this case.

Also - should this be error_logger?

Reply:
This is the False False case, so their script file name was invalid. This means it was changed, not the default